### DISCUSSION TITLE
Fix search URL: searx shows localhost URL instead of public URL

### DISCUSSION THREAD

**Discussion by jsfan3** [searx/searx#3177] [Installation issue: Docker] [answered]
I installed Searx via Docker and then configured Apache as a reverse proxy over http. Then, via Let's Encrypt, I added support for https.

My issue is that the "search URL" exposed to the user in the "Links" box refers to the internal localhost URL instead of the public URL.
I attach a screenshot:

![Schermata del 2022-03-05 10-04-56](https://user-images.githubusercontent.com/1997316/156876568-947e57a0-704c-4dae-b14d-be83e82a1f9b.png)

**Comment by unixfox** [comment, marked as answer]
Either you read what each parameters you are using with docker: https://searx.github.io/searx/admin/installation-docker.html#command-line or you use https://github.com/searx/searx-docker

**Reply by jsfan3** [reply]
Specifically, this is what I use:
```
export PORT=10099
docker run --restart unless-stopped -d -v ${PWD}/searx:/etc/searx -p $PORT:8080 -e BASE_URL=http://localhost:$PORT/ searx/searx
```

and this is my `/etc/apache2/sites-enabled/000-default-le-ssl.conf`:
```
<IfModule mod_ssl.c>
<VirtualHost *:443>
        ServerName ---
        ServerAdmin ---
        #DocumentRoot /var/www/html
        ErrorLog ${APACHE_LOG_DIR}/error.log
        CustomLog ${APACHE_LOG_DIR}/access.log combined
        ProxyRequests Off
        ProxyPreserveHost On
        AddDefaultCharset utf-8
        HostnameLookups off
        UseCanonicalName off
        ProxyBadHeader Ignore
        KeepAlive off
        ProxyPass / http://localhost:10099/
        ProxyPassReverse / http://localhost:10099/


SSLCertificateFile /etc/letsencrypt/live/ricercaalternativa.mydissent.net/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/ricercaalternativa.mydissent.net/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf
</VirtualHost>
</IfModule>
```

At this condition, is there anything I can change to get the correct "search URL"?

Reinstalling searx via https://github.com/searx/searx-docker is not a good option, because I'm using my custom fork of https://github.com/searx/searx (I customized the graphical interface). I cloned that fork and then created the docker image via `make docker.build`.

**Reply by unixfox** [reply]
BASE_URL equals to this parameter in the settings.yml of searx, please read the meaning of it and you will understand what to change: https://github.com/searx/searx/blob/master/searx/settings.yml#L28

**Reply by jsfan3** [reply]
Thank you, it was effortless to fix! :-)
